_posts/2019-08-04-rspec-described_class.markdown

```diff
@@ -0,0 +1,120 @@
+Unfortunately, this was not reflected in the docs ([1](https://relishapp.com/rspec/rspec-core/v/3-8/docs/metadata/described-class), [2](https://relishapp.com/rspec/rspec-core/v/3-8/docs/subject/implicitly-defined-subject)), and it's still not (it's on [`master`](https://github.com/rspec/rspec-core/blob/master/features/metadata/described_class.feature) but [not on `3-8-maintenance`](https://github.com/rspec/rspec-core/blob/3-8-maintenance/features/metadata/described_class.feature)), even after [this pull request](https://github.com/rspec/rspec-core/pull/2629).
+
+> I also argue against using classes there (with describe) preferring strings
+>
+> -- <cite>[Jon Rowe](https://github.com/rspec/rspec-core/issues/1610#issuecomment-46255261)</cite>
+
+```ruby
+# bad
+RSpec.describe Article do
+
+# good
+RSpec.describe “Article” do
+```
+
+> I recommend sticking to `describe <string arg>` for nested example groups as it avoids confusion.
+>
+> -- <cite>[Myron Marston](https://github.com/rspec/rspec-core/issues/1610#issuecomment-46236668)</cite>
+
+> I'd naively expect described_class to return "the class of the described thing" (i.e. Class when describing an actual class). Leaving open because this is definitely still an inconsistency.
+>
+> -- <cite>[Xavier Shay](https://github.com/rspec/rspec-core/issues/1610#issuecomment-272717207)</cite>
+
+There was [an idea](https://github.com/rspec/rspec-core/issues/1610#issuecomment-46316040) to change `described_class`'s behaviour:
+
+```ruby
+describe [1,2,3] do
+  it { described_class.should eq(Array) }
+  it { subject.should be_a(Array) }
+end
+```
+
+Unfortunately, this idea was two weeks late for RSpec 3.0 release.
+
+> I like your idea, @betesh. I wish we had thought of it before 3.0 shipped.
+>
+> -- <cite>[Myron Marston](https://github.com/rspec/rspec-core/issues/1610#issuecomment-46324892)</cite>
+
+With all this confusion in mind:
+
+> I think in RSpec 4 we should just remove described_class, we kind of consider it an anti-pattern (because using it means the class must be loaded when the file is evaluated rather than when it's run) and we thus don't recommend it, and it causes this kind of ambiguity.
+>
+> -- <cite>[Jon Rowe](https://github.com/rspec/rspec-core/issues/1610#issuecomment-515507697)</cite>
+
+### Bottom Line
+
+Pros of `described_class`:
+ - is widely used
+ - reads nicely (when is used properly)
+ - reduces churn if class under test name changes
+
+Cons of `described_class`:
+ - leads to confusion with nested example groups
+ - leads to confusion when metadata is used without docstring
+ - leads to confusion when something that is neither a String and neither a Class is passed to example group as the first argument
+ - might disappear in RSpec 4.0
+
+### How to Deal with This
+
+What do you think about introducing a guideline to discourage passing anything than a literal class and a string as the first argument to an example group?
+React with a :rocket: if you agree.
```

Comment:
Emojis need a special jekyll plugin. (and they don't work well with many themes)

Reply:
Voting is supposed to happen on GitHub, I don't know how to emphasize that even stronger.